### ISSUE TITLE
track unbirthdays too: day of the week + day of month

### ISSUE THREAD

**Issue by travisfw** [m-i-n-a-r/birday#29]
Many people dislike the date of their birthday for some reason, for instance: if they were born on Christmas. An alternative birthday to celebrate is the day of the week plus the day of the month. My grandmother was born on Friday the 13th. She sometimes would celebrate Friday the 13th as though it were a birthday. IIRC, I think I was born on Sunday the 28th, so I could celebrate any month when the 28th is also a Sunday. The term "unbirthday" has been used a number of different ways, but while this isn't the most popular use, I think this is the only well-defined and useful definition of unbirthdays that I have come across.

Unbirthdays can occur up to twice a year, but are also absent some years, and also sometimes fall on the birthday itself. This is the kind of thing it would be helpful to rely on an app to track.

**Comment by m-i-n-a-r** [comment]
Never heard of this unbirthday thing. I could implement something similar in the stats, what worries me is that many people won't understand the meaning of "unbirthday". Sorry for the late reply though, i'll see what i can do!

**Comment by travisfw** [comment]
> what worries me is that many people won't understand the meaning of "unbirthday".
Unfortunately, you're very probably right. Although I certainly did not originate this definition of "unbirthday", I've explained it a lot too.

**Comment by m-i-n-a-r** [comment]
Yep i can think of something similar, i have to explain it. I'll try to find a smart solution 👍

**Comment by DominikNovosel** [comment]
Perhaps someone should maintain a separate fork of Birday called unBirday. 😆 It seems to me that this functionality would require a relatively big refactor of its existing features and interface to the point of interfering with its original function and causing confusion to regular users. While we're at it, it could be handy to visualize "unbirthday opportunities" by highlighting days on a calendar that match a set of criteria such as you described.

**Comment by m-i-n-a-r** [comment]
Yep, you're right. I'm still trying to satisfy this request, but it's quite hard to find a smart and easy way to implement such feature, especially considering that i want to keep the app simple to use!

**Comment by DominikNovosel** [comment]
I think I managed to come up with a partial solution (complete with a snazzy concept for an icon!):
- When inputting a new birthday, we ask whether there are special demands as to when the birthday is celebrated.

![Unbirday UI concept art1](https://user-images.githubusercontent.com/66269443/149635413-17f197eb-4a7d-4d58-9327-69d0e9b6ef5c.png)

- We also add an info icon that can pop up a text box providing an explanation for what this does.

![Unbirday UI concept art2](https://user-images.githubusercontent.com/66269443/149635547-1d848daa-126d-4c47-af26-8fdf645b9edd.png)

The only real issues I can see is that the term "unbirthday" has no widely accepted definition and its behaviour might be a bit difficult to implement.
Fun fact: the term "unbirthday" actually originated from L. Carrol's "Through the Looking-Glass", where it behaves [completely nonsensically](https://en.wikipedia.org/wiki/Unbirthday).
Also, I intended for the dotted outline in the icon to be dashed instead of dotted, but I didn't want to spend too much time on this.
Here's the source file:
[Unbirday UI concept art.zip](https://github.com/m-i-n-a-r/birday/files/7875758/Unbirday.UI.concept.art.zip)

**Comment by m-i-n-a-r** [comment]
Well that's an interesting solution for sure, the reason why i gave up on this issue is the same you wrote above, too much translations involved, and the many people don't know what an unbirthday is, including many people i know, and me before this issue was opened 😄

**Comment by DominikNovosel** [comment]
Thanks! It has been suggested to me recently that we shouldn't clutter the birthday input form with an option most people don't care about and that this should be determined in the settings per birthday, but that sounds like it would be a pain to use. I think a good compromise would be to add a setting that toggles whether this option is shown, we could even ask about the user's familiarity with unbirthdays on the first startup of the app to set the value immediately.
